packages/expo-modules-autolinking/src/autolinking.ts

```diff
@@ -62,84 +62,203 @@ export async function findDefaultPathsAsync(cwd: string): Promise<string[]> {
+  if (nativeModulesDir) {
+    const nativeModulesDirPath = path.resolve(nativeModulesDir);
+    if (await fs.pathExists(nativeModulesDirPath)) {
+      return nativeModulesDirPath;
+    }
+  }
+
+  // if not found, try to find it relative to the package.json
+  const up = await findUp('package.json', { cwd });
+  if (!up) {
+    return undefined;
+  }
+  const resolvedPath = path.join(up, '..', nativeModulesDir || 'modules');
+  return fs.existsSync(resolvedPath) ? resolvedPath : undefined;
+}
+
+/**
+ * Adds {@link revision} to the {@link results} map
+ * or to package duplicates if it already exists.
+ * @param results [mutable] yet resolved packages map
+ * @param name resolved package name
+ * @param revision resolved package revision
+ */
+function addRevisionToResults(
+  results: Map<string, PackageRevision>,
+  name: string,
+  revision: PackageRevision
+): void {
+  if (!results.has(name)) {
+    // The revision that was found first will be the main one.
+    // An array of duplicates and the config are needed only here.
+    results.set(name, {
+      ...revision,
+      duplicates: [],
+    });
+  } else if (
+    results.get(name)?.path !== revision.path &&
+    results.get(name)?.duplicates?.every(({ path }) => path !== revision.path)
+  ) {
+    const { config, duplicates, ...duplicateEntry } = revision;
+    results.get(name)?.duplicates?.push(duplicateEntry);
+  }
+}
+
+/**
+ * Returns paths to the highest priority config files, relative to the {@link searchPath}.
+ * @example
+ * ```
+ * // Given the following file exists: /foo/myapp/modules/mymodule/expo-module.config.json
+ * await findPackagesConfigPathsAsync('/foo/myapp/modules');
+ * // returns ['mymodule/expo-module.config.json']
+ * ```
+ */
+async function findPackagesConfigPathsAsync(searchPath: string): Promise<string[]> {
+  const bracedFilenames = '{' + EXPO_MODULE_CONFIG_FILENAMES.join(',') + '}';
+  const paths = await glob([`*/${bracedFilenames}`, `@*/*/${bracedFilenames}`], {
+    cwd: searchPath,
+  });
+
+  // If the package has multiple configs (e.g. `unimodule.json` and `expo-module.config.json` during the transition time)
+  // then we want to give `expo-module.config.json` the priority.
+  return Object.values(
+    paths.reduce<Record<string, string>>((acc, configPath) => {
+      const dirname = path.dirname(configPath);
+
+      if (!acc[dirname] || configPriority(configPath) > configPriority(acc[dirname])) {
+        acc[dirname] = configPath;
+      }
+      return acc;
+    }, {})
+  );
+}
+
+/**
+ * Resolves package name and version for the given {@link packagePath} from its `package.json`.
+ * if {@link options.fallbackToDirName} is true, it returns the dir name when `package.json` doesn't exist.
+ * @returns object with `name` and `version` properties. `version` falls back to `UNVERSIONED` if cannot be resolved.
+ */
+function resolvePackageNameAndVersion(
+  packagePath: string,
+  options: { fallbackToDirName?: boolean } = {}
```

Comment:
As there is just one option, maybe it makes more sense to "unwrap" it? 😄

Reply:
maybe yes, I don't usually do that, because I hate TS syntax for that. You have to define options twice. Wish it was sth like this:
```ts
function resolvePackageNameAndVersion(
  packagePath: string,
  { fallbackToDirName?: boolean } = {}
) {
```